### ISSUE TITLE
Different results when training for 2 epochs vs training 1 epoch, stoping and training again for another epoch.

### ISSUE THREAD

**Issue by bristena-op** [Lightning-AI/lightning#13058]
## 🐛 Bug

When stopping and re-starting training on a PyTorch model using the RAdam optimizer, results in different state dictionaries for the model. Is there any reason that this works likes this? Stopping and restarting training is very useful for a number of different applications (e.g. transfer learning, federated learning, etc.), and would be important to be able to get the same results when training in one go and whenst

### To Reproduce
Updated here: https://colab.research.google.com/drive/1TYcGOK3nu7N9Ys6ZcF-PFejkqfwbsMLF?usp=sharing

### Expected behavior

Same model state_dict after training 2 epochs or training 1 epoch, toping and starting again for on more epoch.
### Environment

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           11.3
* Packages:
	- numpy:             1.21.6
	- pyTorch_debug:     False
	- pyTorch_version:   1.11.0+cu113
	- pytorch-lightning: 1.6.3
	- tqdm:              4.64.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.7.13
	- version:           #1 SMP Sun Apr 24 10:03:06 PDT 2022

### Additional context

cc @awaelchli @ananthsub @ninginthecloud @rohitgr7 @otaj

**Comment by tanmoyio** [comment]
Taking a look, I haven't seen your notebook yet, but a blind guess,  it might be related to lr scheduler. Not sure, working on it.

**Comment by rohitgr7** [comment]
when resuming the training, only model weights are not necessary to be reloaded. In a simple case like yours optimizer states needs to be reloaded too.
so
```py
model3.load_state_dict(model2.state_dict())
trainer3.fit(model3, train_dataloaders=train_data, val_dataloaders=val_data)
```
this is not enough.

I'd suggest doing:
```py
ckpt_path = trainer2.checkpoint_callback.last_model_path
model3 = BoringModel().to(device)
trainer3 = Trainer(
    max_epochs=1,
    enable_model_summary=False,
    deterministic=True
)

trainer3.fit(model3, train_dataloaders=train_data, val_dataloaders=val_data, ckpt_path=ckpt_path)
```
